domain/application/state/application.go

```diff
@@ -1733,10 +1733,119 @@ INSERT INTO cloud_service (*) VALUES ($cloudService.*)
 	return nil
 }
 
+// GetApplicationStatus looks up the status of the specified application,
+// returning an error satisfying [applicationerrors.ApplicationNotFound] if the
+// application is not found.
+func (st *State) GetApplicationStatus(ctx context.Context, appID coreapplication.ID) (*application.StatusInfo[application.WorkloadStatusType], error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, jujuerrors.Trace(err)
+	}
+
+	identID := applicationID{ID: appID}
+	identUUID := applicationUUID{ApplicationUUID: appID.String()}
+
+	query, err := st.Prepare(
+		`
+SELECT &applicationStatusInfo.*
+FROM application_status
+WHERE application_uuid = $applicationUUID.application_uuid;
+`, identUUID, applicationStatusInfo{})
+	if err != nil {
+		return nil, jujuerrors.Trace(err)
+	}
+
+	var status applicationStatusInfo
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		if err := st.checkApplicationExists(ctx, tx, identID); err != nil {
+			return jujuerrors.Trace(err)
+		}
+
+		if err := tx.Query(ctx, query, identUUID).Get(&status); errors.Is(err, sqlair.ErrNoRows) {
+			return nil
+		} else if err != nil {
+			return jujuerrors.Trace(err)
+		}
+
+		return nil
+	})
```

Comment:
This is not how `GetUnitWorkloadStatus` behaves, but this is probably correct

`GetWorkloadUnitStatus` throws a `applicationerrors.UnitStatusNotFound` err if there is no unit status, which results either from the unit not existing, or it having no status

```go
	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
		err := tx.Query(ctx, getUnitStatusStmt, unitUUID).Get(&unitStatusInfo)
		if errors.Is(err, sql.ErrNoRows) {
			return errors.Errorf("workload status for unit %q not found%w", unitUUID, jujuerrors.Hide(applicationerrors.UnitStatusNotFound))
		}
		return err
	})
```

Reply:
Applications are different. If there is no status, it returns not set. This is by design.